packages/fxa-event-broker/lib/db/firestore.ts

```diff
@@ -20,29 +22,36 @@ class FirestoreDatastore implements Datastore {
     const document = this.db.doc(`users/${uid}`);
     const doc = await document.get();
     if (doc.exists) {
-      const data: any = doc.data();
-      if (data.oauth_clients && data.oauth_clients[clientId]) {
+      const data = doc.data();
+      if (data && data.oauth_clients && data.oauth_clients[clientId]) {
         // Record is already in the database
         return;
       }
     }
-    await document.set(
-      { oauth_clients: { [clientId]: true } },
-      { merge: true }
-    );
+    await document.set({ oauth_clients: { [clientId]: true } }, { merge: true });
   }
 
   public async fetchClientIds(uid: string): Promise<string[]> {
     const document = this.db.doc(`users/${uid}`);
     const doc = await document.get();
     if (doc.exists) {
-      const data: any = doc.data();
-      if (data.oauth_clients) {
+      const data = doc.data();
+      if (data && data.oauth_clients) {
```

Comment:
Google API's have types already, so casting to any lost important type data, such as doc.data() possibly returning undefined, which is now tested for.